packages/cli/src/dev.ts

```diff
@@ -82,6 +82,10 @@ export async function entrypointHasDefaultExport(
   return false;
 }
 
+function replaceTsExtensionWithJs(pathname: string) {
+  return pathname.replace(/\.(?:tsx|([mc])?ts)$/, ".$1js");
```

Comment:
nit: as far as I can see `.cts` and `.mts` cases are not covered with tests ;p

Reply:
Yeah, mentioned those extensions in the PR description, I believe they just don't work right now and fixing that isn't really a high priority tbh since they aren't really useful in Preconstruct's context and we'd want to ban mixing module types anyway